examples/C++/DynamicHelloWorldExample/HelloWorldPublisher.cpp

```diff
@@ -0,0 +1,251 @@
+ * @file HelloWorldPublisher.cpp
+ *
+ */
+
+#include "HelloWorldPublisher.h"
+#include "HelloWorldTypeObject.h"
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/PublisherAttributes.h>
+#include <fastrtps/publisher/Publisher.h>
+#include <fastrtps/Domain.h>
+#include <fastrtps/utils/eClock.h>
+#include <fastrtps/types/DynamicTypeBuilderFactory.h>
+#include <fastrtps/types/DynamicDataFactory.h>
+#include <fastrtps/types/DynamicTypeBuilder.h>
+#include <fastrtps/types/DynamicTypeBuilderPtr.h>
+#include <fastrtps/types/TypeDescriptor.h>
+#include <fastrtps/types/MemberDescriptor.h>
+#include <fastrtps/types/DynamicType.h>
+
+using namespace eprosima::fastrtps;
+using namespace eprosima::fastrtps::rtps;
+
+HelloWorldPublisher::HelloWorldPublisher()
+    : mp_participant(nullptr)
+    , mp_publisher(nullptr)
+    , m_DynType(nullptr)
+    , m_dynamic(false)
+{
+    m_Hello = nullptr;
+}
+
+bool HelloWorldPublisher::init(bool dynamic)
+{
+    m_dynamic = dynamic;
+    if (dynamic)
+    {
+        // Create basic builders
+        DynamicTypeBuilder_ptr struct_type_builder(DynamicTypeBuilderFactory::GetInstance()->CreateStructBuilder());
+
+        // Add members to the struct.
+        struct_type_builder->AddMember(0, "index", DynamicTypeBuilderFactory::GetInstance()->CreateUint32Type());
+        struct_type_builder->AddMember(1, "message", DynamicTypeBuilderFactory::GetInstance()->CreateStringType());
+        struct_type_builder->SetName("HelloWorld");
+
+        DynamicType_ptr dynType = struct_type_builder->Build();
+        m_DynType.SetDynamicType(dynType);
+        m_DynHello = DynamicDataFactory::GetInstance()->CreateData(dynType);
+        m_DynHello->SetUint32Value(0, 0);
+        m_DynHello->SetStringValue("HelloWorld", 1);
+    }
+    else
+    {
+        m_Hello = new HelloWorld;
+        m_Hello->index(0);
+        m_Hello->message("HelloWorld");
+    }
+
+    ParticipantAttributes PParam;
+    PParam.rtps.builtin.use_SIMPLE_RTPSParticipantDiscoveryProtocol = true;
+    PParam.rtps.builtin.use_SIMPLE_EndpointDiscoveryProtocol = true;
+    PParam.rtps.builtin.m_simpleEDP.use_PublicationReaderANDSubscriptionWriter = true;
+    PParam.rtps.builtin.m_simpleEDP.use_PublicationWriterANDSubscriptionReader = true;
+    PParam.rtps.builtin.domainId = 5;
+    PParam.rtps.builtin.leaseDuration = c_TimeInfinite;
+    PParam.rtps.setName("DynHelloWorld_pub");
+    mp_participant = Domain::createParticipant(PParam, (ParticipantListener*)&m_part_list);
+
+    if(mp_participant==nullptr)
+        return false;
+    //REGISTER THE TYPE
+
+    if (m_dynamic)
+    {
+        Domain::registerDynamicType(mp_participant, &m_DynType);
+    }
+    else
+    {
+        Domain::registerType(mp_participant, &m_type);
+    }
+
+    //CREATE THE PUBLISHER
+    PublisherAttributes Wparam;
+    Wparam.topic.topicKind = NO_KEY;
+    Wparam.topic.topicDataType = "HelloWorld";
+    Wparam.topic.topicName = "DynamicHelloWorldTopic";
```

Comment:
Use same topicName as normal HelloWorldExample. Doing this both examples can interoperate.